llama-index-integrations/vector_stores/llama-index-vector-stores-azurecosmosnosql/llama_index/vector_stores/azurecosmosnosql/base.py

```diff
@@ -325,25 +325,31 @@ def _query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResu
 
         # If limit_offset_clause is not specified, add TOP clause
         if pre_filter is None or pre_filter.get("limit_offset_clause") is None:
-            query += "TOP @limit "
+            # query += "TOP @limit "
+            query += f"TOP {params.get('k', 2)} "
```

Comment:
Also curious, can you give some explanation on this change? This seems to be changing a lot of the query text